content/learn/opaque-types-performance.md

```diff
@@ -0,0 +1,67 @@
+---
+type: page
+title: A case for opaque types
+description: The performance benefits of defaulting to opaque types
+published: false
+status: seedling
+authors: "@aemengo"
+editors: ""
+---
+
+[Opaque types](https://sporto.github.io/elm-patterns/advanced/opaque-types.html) as an Elm design pattern always made sense - for libraries. The idea being that I use a **union type** for a plain old Elm object, instead of an **alias type**, even when there's only one **variant**.
```

Comment:
My feeling (but feel free to disregard it) is that the post uses maybe an unoptimal route for explaining the problem and the solution, going through explaining opaque types which is not the main concern. You (and the people who will be sent this link) have a **compilation** performance issue which is resolved through custom types.

I think it would be more to the point to change the article to talk about potential issues around compilation speeds, things to look for, and solutions. I would really emphasize **compilation** speeds too, and maybe not talk about design and custom types which I would say are a bit beside the point.

I think the style chosen for this article is also more of a blog post (see the uses of the first pronoun and the usage of terms like "assume") rather than documentation where people are looking for solutions to their problems (See for example this article for the style I'm thinking of: https://elmcraft.org/compare/javascript/eslint).